File: jscomp/test/bs_map_test.js
'use strict';

var Mt = require("./mt.js");
var Belt_Array = require("../../lib/js/belt_Array.js");
var Belt_MapInt = require("../../lib/js/belt_MapInt.js");
var Belt_SetInt = require("../../lib/js/belt_SetInt.js");

var suites = {
  contents: /* [] */0
};

var test_id = {
  contents: 0
};

function eq(loc, x, y) {
  test_id.contents = test_id.contents + 1 | 0;
  suites.contents = {
    hd: [
      loc + (" id " + String(test_id.contents)),
      (function (param) {
          return {
                  TAG: /* Eq */0,
                  _0: x,
                  _1: y
                };
        })
    ],
    tl: suites.contents
  };
  
}

function b(loc, v) {
  test_id.contents = test_id.contents + 1 | 0;
  suites.contents = {
    hd: [
      loc + (" id " + String(test_id.contents)),
      (function (param) {
          return {
                  TAG: /* Ok */4,
                  _0: v
                };
        })
    ],
    tl: suites.contents
  };
  
}

var mapOfArray = Belt_MapInt.fromArray;

var setOfArray = Belt_SetInt.fromArray;

function emptyMap(param) {
  
}

var v = Belt_Array.makeByAndShuffle(1000000, (function (i) {
        return [
                i,
                i
              ];
      }));

var u = Belt_MapInt.fromArray(v);

Belt_MapInt.checkInvariantInternal(u);

var firstHalf = Belt_Array.slice(v, 0, 2000);

var xx = Belt_Array.reduce(firstHalf, u, (function (acc, param) {
        return Belt_MapInt.remove(acc, param[0]);
      }));

Belt_MapInt.checkInvariantInternal(u);

Belt_MapInt.checkInvariantInternal(xx);

Mt.from_pair_suites("Bs_map_test", suites.contents);

var M;

var N;

var A;

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
exports.b = b;
exports.M = M;
exports.N = N;
exports.A = A;
exports.mapOfArray = mapOfArray;
exports.setOfArray = setOfArray;
exports.emptyMap = emptyMap;
/* v Not a pure module */

File: jscomp/test/bs_map_test.ml
let suites :  Mt.pair_suites ref  = ref []
let test_id = ref 0
let eq loc x y = 
  incr test_id ; 
  suites := 
    (loc ^" id " ^ (string_of_int !test_id), (fun _ -> Mt.Eq(x,y))) :: !suites

let b loc v  = 
  incr test_id ; 
  suites := 
    (loc ^" id " ^ (string_of_int !test_id), 
     (fun _ -> Mt.Ok v)) :: !suites

module M = Belt.Map.Int
module N = Belt.Set.Int
module A = Belt.Array


let mapOfArray x = M.fromArray x 
let setOfArray x = N.fromArray  x 
let emptyMap () = M.empty 

let () = 
  let v = 
      (A.makeByAndShuffle 1_000_000 (fun i -> (i,i))) in 
  let u = M.fromArray v in   
   (M.checkInvariantInternal u);
  let firstHalf = A.slice v 0 2_000 in 
  let xx = A.reduce firstHalf u
      (fun acc (x,_) -> M.remove acc x)  in 
  (M.checkInvariantInternal u);
  (M.checkInvariantInternal xx);


;; Mt.from_pair_suites __FILE__ !suites
